zipline/testing/predicates.py

```diff
@@ -552,6 +552,14 @@ def assert_array_equal(result,
         if array_decimal is None else
         partial(np.testing.assert_array_almost_equal, decimal=array_decimal)
     )
+
+    if result.dtype.kind == 'M' and expected.dtype.kind == 'M':
```

Comment:
if only one of the arrays has 'M' dtype, we can let `assert_array_equal` handle the mismatch.

Reply:
I would either use `.view` here or call `astype` with `copy=False`. The default behavior for `astype` is to copy the data, which is unnecessary here.